NorthstarDLL/structs.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+//clang-format off
+// About this file:
+// This file contains several macros used to define reversed structs
+// The reason we use these macros is to make it easier to update existing structs
+// when new fields are reversed
+// This means we dont have to manually add padding, and recalculate when updating
+
+// Technical note:
+// While functionally, these structs act like a regular struct, they are actually
+// defined as unions with anonymous structs in them.
+// This means that each field is essentially an offset into a union.
+// We acknowledge that this goes against C++'s active-member guideline for unions
+// However, this is not such a big deal here as these structs will not be constructed
+
+// Usage:
+// To use these macros, define a struct like so:
+/*
+OFFSET_STRUCT(Name)
+{
+	STRUCT_SIZE(0x100)		// Total in-memory struct size
+	FIELD(0x0, int field)	// offset, signature
+}
+*/
+
+#define OFFSET_STRUCT(name) union name
```

Comment:
debating whether this should be called something like INTERNAL_STRUCT instead, but unsure

Reply:
Well its not really internal or anything. OFFSET_STRUCT describes what it is: a struct defined with offsets

Reply:
yeah fair